data/abilities.js

```diff
@@ -878,7 +878,7 @@ exports.BattleAbilities = {
 			}
 			if (!warnMoves.length) return;
 			var warnMove = warnMoves[this.random(warnMoves.length)];
-			this.add('-activate', pokemon, 'ability: Forewarn', warnMove[0], warnMove[1]);
+			this.add('-activate', pokemon, 'Forewarn', warnMove[0], '[of] ' + warnMove[1]);
```

Comment:
This should be `ability: Forewarn`, not `Forewarn`. If you don't put any tag in front of an effect, it's a volatile status.